velox/type/tz/TimeZoneMap.cpp

```diff
@@ -15,14 +15,96 @@
  */
 
 #include "velox/type/tz/TimeZoneMap.h"
+#include <boost/algorithm/string.hpp>
 #include <fmt/core.h>
 #include <folly/container/F14Map.h>
+#include <folly/container/F14Set.h>
 
 namespace facebook::velox::util {
 
 // Defined on TimeZoneDatabase.cpp
 extern const std::unordered_map<int64_t, std::string>& getTimeZoneDB();
 
+namespace {
+
+folly::F14FastMap<std::string, int64_t> makeReverseMap(
+    const std::unordered_map<int64_t, std::string>& map) {
+  folly::F14FastMap<std::string, int64_t> reversed;
+  reversed.reserve(map.size() + 1);
+
+  for (const auto& entry : map) {
+    reversed.emplace(
+        boost::algorithm::to_lower_copy(entry.second), entry.first);
+  }
+  reversed.emplace("utc", 0);
+  return reversed;
+}
+
+inline bool isDigit(char c) {
+  return c >= '0' && c <= '9';
+}
+
+inline bool startsWith(std::string_view str, const char* prefix) {
+  return str.rfind(prefix, 0) == 0;
+}
+
+inline bool isUtcEquivalentName(std::string_view zone) {
+  static folly::F14FastSet<std::string> utcSet = {
+      "utc", "uct", "gmt", "gmt0", "greenwich", "universal", "zulu", "z"};
```

Comment:
Curious where does this list come from?

Reply:
The list is here.:

https://en.wikipedia.org/wiki/List_of_tz_database_time_zones

I'll add a link to the code and docs.